### PR TITLE
Updated ResNet to work with 2019-06-04 development snapshot.

### LINKED ISSUE #165: Rename 'call' to 'callAsFunction'.
`Layer.call(_:)` has been renamed to `Layer.callAsFunction(_:)` (https://github.com/tensorflow/swift-apis/pull/172) as the result of [SE-0253](https://github.com/apple/swift-evolution/blob/master/proposals/0253-callable.md) proposal review. A toolchain with the new `Layer` definition is [ready for download](https://github.com/tensorflow/swift/blob/master/Installation.md#development-snapshots).

Task: Update all models and layers in this repo to define `func callAsFunction` instead of `func call`.

### PR DESCRIPTION
**Pull request by leoxzhao**
Partially fixes #165.

### CODE REVIEW

File: ResNet/Data.swift [hunk 1]
```diff
@@ -35,6 +35,25 @@ func downloadCIFAR10IfNotPresent(to directory: String = ".") {
...
+        _handles: C
+    ) where C.Element: _AnyTensorHandle {
+        precondition(_handles.count == 2)
+        let labelIndex = _handles.startIndex
+        let dataIndex = _handles.index(labelIndex, offsetBy: 1)
+        label = Tensor<Int32>(handle: TensorHandle<Int32>(handle: _handles[labelIndex]))
+        data = Tensor<Float>(handle: TensorHandle<Float>(handle: _handles[dataIndex]))
+    }
+
+    //public var _tensorHandles: [_AnyTensorHandle] { [label.handle.handle, data.handle.handle] }
```
Comment (leoxzhao) [inline ResNet/Data.swift]: I feel this is a bug, but not entirely sure. How else to satisfy TensorGroup protocol?

Comment (rxwei) [reply ResNet/Data.swift]: This is a known regression introduced in https://github.com/tensorflow/swift-apis/pull/165 and should be fixed soon.

**Comment by leoxzhao** [comment]
I have incorporated this change into https://github.com/tensorflow/swift-models/pull/166 so that one PR updates all models. Close this PR.

**Comment by rxwei** [comment]
Sounds good!

**Pull request closed without merging**